paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PrimaryButtonNew.kt

```diff
@@ -0,0 +1,381 @@
+        background
+    } else {
+        animateColorAsState(
+            targetValue = if (isProcessingCompleted) {
+                processCompleteBackground
+            } else {
+                Color(style.getBackgroundColor(context))
+            },
+            label = "BackgroundAnimation",
+            animationSpec = tween(durationMillis = 100, easing = LinearEasing)
+        ).value
+    }
+
+    val borderStroke = BorderStroke(style.shape.borderStrokeWidth.dp, Color(style.getBorderStrokeColor(context)))
+    val shape = RoundedCornerShape(style.shape.cornerRadius.dp)
+
+    CompositionLocalProvider(
+        LocalContentAlpha provides if (enabled) ContentAlpha.high else ContentAlpha.disabled,
+    ) {
+        Box(
+            modifier = Modifier.fillMaxWidth(),
+            contentAlignment = Alignment.Center,
+        ) {
+            TextButton(
+                onClick = onClick,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .defaultMinSize(
+                        minHeight = dimensionResource(id = R.dimen.stripe_paymentsheet_primary_button_height)
+                    ),
+                enabled = enabled,
+                shape = shape,
+                border = borderStroke,
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = animatedBackground,
+                    disabledBackgroundColor = animatedBackground,
+                ),
+            ) {
+                Content(
+                    label = label,
+                    processingState = processingState,
+                    enabled = enabled,
+                    locked = locked,
+                    onProcessingCompleted = onProcessingCompleted,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun Content(
+    label: String,
+    processingState: PrimaryButtonProcessingState,
+    enabled: Boolean,
+    locked: Boolean,
+    onProcessingCompleted: () -> Unit,
+) {
+    AnimatedContent(
+        processingState is PrimaryButtonProcessingState.Completed,
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(IntrinsicSize.Max),
+        transitionSpec = {
+            fadeIn(animationSpec = tween(FADE_ANIMATION_DURATION)) togetherWith
+                fadeOut(animationSpec = tween(FADE_ANIMATION_DURATION, FADE_OUT_ANIMATION_DELAY))
+        },
+        label = "ContentAnimation",
+    ) { completed ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = 8.dp)
+        ) {
+            if (completed) {
+                Complete(onProcessingCompleted)
+            } else {
+                Incomplete(
+                    text = when (processingState) {
+                        is PrimaryButtonProcessingState.Idle -> label
+                        else -> stringResource(R.string.stripe_paymentsheet_primary_button_processing)
+                    },
```

Comment:
We should work to have the `Processing…` text come as the label instead of setting it internally, but this is fine for now.